jitbuilder/compile/Method.hpp

```diff
@@ -166,7 +167,7 @@ class ResolvedMethod : public ResolvedMethodBase, public Method
                                TR::SymbolReferenceTable *symRefTab);
 
    protected:
-   const char *_fileName;
+   std::string _fileName;
```

Comment:
I missed that this was on Method.hpp -- so I agree, it either needs to not use `std::string` or it needs to use an allocator to do its allocation. 

@sajidahmed21 asked if a MethodBuilder can be expected to outlive `ResolvedMethod` ... I think the answer is yes here.